csv2vcf.py

```diff
@@ -10,28 +10,51 @@
 import csv
 import json
 
+default_input_file_format = {
+    'first_name': None,
+    'last_name': None,
+    'full_name': None,
+    'nickname': None,
+    'org': None,
+    'tel': None,
+    'url': None,
+    'bday': None,
+    'role': None,
+    'email': None,
+    'note': None
+}
+
 
 def convert_to_vcard(input_file, single_output, input_file_format):
 
-    FN = input_file_format['name']-1 if 'name' in input_file_format else None
-    NICKNAME = input_file_format['nickname']-1 if 'nickname' in input_file_format else None
-    ORG = input_file_format['org']-1 if 'org' in input_file_format else None
-    TEL = input_file_format['tel']-1 if 'tel' in input_file_format else None
-    URL = input_file_format['url']-1 if 'url' in input_file_format else None
-    BDAY = input_file_format['bday']-1 if 'bday' in input_file_format else None
-    ROLE = input_file_format['role']-1 if 'role' in input_file_format else None
-    EMAIL = input_file_format['email']-1 if 'email' in input_file_format else None
-    NOTE = input_file_format['note']-1 if 'note' in input_file_format else None
+    input_file_format = {**default_input_file_format, **input_file_format}
+
+    FIRST_NAME = input_file_format['first_name']
+    LAST_NAME = input_file_format['last_name']
+    FULL_NAME = input_file_format['full_name']
+
+    NICKNAME = input_file_format['nickname']
+
+    ORG = input_file_format['org']
+    TEL = input_file_format['tel']
+    URL = input_file_format['url']
+    BDAY = input_file_format['bday']
+    ROLE = input_file_format['role']
+    EMAIL = input_file_format['email']
+    NOTE = input_file_format['note']
 
     # if single output option is selected
-    if single_output :
-        with open( input_file, 'r' ) as source_file:
-            reader = csv.reader( source_file )
+    if single_output:
+        with open(input_file, 'r', encoding='utf-8-sig') as source_file:
+            reader = csv.DictReader(source_file)
```

Comment:
The `csv` file generated by Outlook contains tons keys:
```
First Name | Middle Name | Last Name | Title | Suffix | Nickname | Given Yomi | Surname Yomi | E-mail Address | E-mail 2 Address | E-mail 3 Address | Home Phone | Home Phone 2 | Business Phone | Business Phone 2 | Mobile Phone | Car Phone | Other Phone | Primary Phone | Pager | Business Fax | Home Fax | Other Fax | Company Main Phone | Callback | Radio Phone | Telex | TTY/TDD Phone | IMAddress | Job Title | Department | Company | Office Location | Manager's Name | Assistant's Name | Assistant's Phone | Company Yomi | Business Street | Business City | Business State | Business Postal Code | Business Country/Region | Home Street | Home City | Home State | Home Postal Code | Home Country/Region | Other Street | Other City | Other State | Other Postal Code | Other Country/Region | Notes | Personal Web Page | Spouse | Schools | Hobby | Location | Web Page | Birthday | Anniversary
```
Indexing by key titles is more convenient



```